packages/aws-cdk-lib/aws-applicationautoscaling/lib/step-scaling-policy.ts

```diff
@@ -117,8 +117,22 @@ export class StepScalingPolicy extends Construct {
       throw new Error(`'scalingSteps' can have at most 40 steps, got ${props.scalingSteps.length}`);
     }
 
-    if (props.datapointsToAlarm !== undefined && props.datapointsToAlarm < 1) {
-      throw new RangeError(`datapointsToAlarm cannot be less than 1, got: ${props.datapointsToAlarm}`);
+    if (props.evaluationPeriods !== undefined && !cdk.Token.isUnresolved(props.evaluationPeriods) && props.evaluationPeriods < 1) {
+      throw new RangeError(`evaluationPeriods cannot be less than 1, got: ${props.evaluationPeriods}`);
+    }
+    if (props.datapointsToAlarm !== undefined) {
+      if (props.evaluationPeriods === undefined) {
+        throw new RangeError('evaluationPeriods must be set if datapointsToAlarm is set');
```

Comment:
```suggestion
        throw new Error('evaluationPeriods must be set if datapointsToAlarm is set');
```
Should be a simple `Error`.
In general, I don't see `RangeError`s being used much in the repo.
I'm wondering if it wouldn't just be better to use the `Error` class for all guards in this file as well 🤔

Reply:
OK, I have adapted the existing code as it is not the essence of this PR, but I will change everything to `Error`.